subtitles.md

```diff
@@ -677,3 +677,36 @@ in segment 7.2 "Syntax and semantics of the subtitling segment" of ETSI EN 300 7
 Each Matroska Block **SHOULD** have a Duration indicating how long the DVB Subtitle Segments
 in that Block **SHOULD** be displayed.
 
+## ARIB (ISDB) subtitles
+
+The specifications for the ARIB B-24 subtitle bitstream format (short: ARIB subtitles)
+and its storage in MPEG transport streams can be found in the documents
+"ARIB STD-B24 - Data Coding and Transmission Specification for Digital Broadcasting",
+"ARIB STD-B10 - Service Information for Digital Broadcasting System",
+and "ARIB TR-B14 - Operational Guidelines for Digital Terrestrial Television Broadcasting".
+
+### Storage of ARIB subtitles
+
+#### CodecID
+
+The CodecID to use is `S_ARIBSUB`.
+
+#### CodecPrivate
+
+The CodecPrivate element is three bytes long and has the following structure:
+
+*    1 byte: component tag (bit string, left bit first)
+*    2 bytes: data component ID (bit string, left bit first)
+
+The semantics of the component tag are the same as those described in ARIB STD-B10, part 2, Annex J.
+The semantics of the data component ID are the same as those described in ARIB TR-B14, fascicle 2, Vol. 3, Section 2, 4.2.8.1.
+
+#### Storage of ARIB subtitles in Matroska Blocks
+
+Each Matroska Block consists of one or more ISDB Caption Data Groups as described
+in chapter 9 "Transmission of caption and superimpose" of ARIB STD-B24 volume 1 part 3.
+
+A Data Group is normally shown until a subsequent Group provides instructions to clear it.
+Therefore the Matroska Block **SHOULD** have no Duration.
+A player **SHOULD** display a Data Group within a Matroska Block until its internal duration elapses,
+or until a subsequent Data Group removes it.
```

Comment:
The problem is that Matroska on its own does not know about internals of codec, and should NOT. If you want to handle cutting files properly, that means relying on a Block duration. In that case the group removal MUST be translated to a Matroska duration. It's up to the player to transform it back to an actual format an ARIB B-24 parser understands, if necessary.

In that case there's no need for this sentence at all since it falls back to the general rule of Block duration.

Reply:
"Removing it" generally means that a later packet includes a command that clears all previous lines from the screen. Most groups don't have a defined duration at all. See also the PGS section of this file, which specifies that "A Segment is normally shown until a subsequent Segment is encountered."

Reply:
The PGS description has more details on what to do when the muxer chooses not put a duration. It is bogus though "a player MUST display a Segment within a Matroska Block until the next Segment is encountered.". The Segment boundaries are arbitrary, a remuxing of the same file will produce a different playback experience. I think the text should be "PGS Segment" and then it makes sense.

Then there's a proper explanation on what a "SHOULD have a duration" means, when there is one and when there isn't one. At least you should have something similar.

It's still equally broken when there is no duration. If you cut a portion of audio/video that also contained the Clear command, then your subtitle can be displayed for hours.

I understand it's easier to mux without changing anything of the original bits, but it produces bogus Matroska files. It would also be hard to find the duration preemptively without preparsing the original source first, or editing the duration of previous blocks while muxing. I don't know of any framework capable of doing that easily.

Reply:
Not sure of mkvmerge, maybe it can preparse files to get a list of subtitle entries and their duration (can be done with PGS as well). cc @mbunkus 

I don't think libavformat would be capable of that.

Reply:
> It would also be hard to find the duration preemptively without preparsing the original source first, or editing the duration of previous blocks while muxing. I don't know of any framework capable of doing that easily.

Yeah, that's the thing; if I required a duration here, or explicitly said that packets without specified durations always end with the next packet, I'd be making it dramatically harder to simply remux (a common operation) in favor of making it simpler to clip and concatenate (a rarer operation). It would also be impossible to do in realtime, which is a requirement for some matroska applications.

Note that all of these issues also apply to ATSC closed captions, though those currently exist in Matroska largely by being carried within MPEG2, H.264, or HEVC bitstreams (e.g. as SEIs).

Reply:
> Not sure of mkvmerge, maybe it can preparse files to get a list of subtitle entries and their duration (can be done with PGS as well). cc @mbunkus

`mkvmerge` cannot pre-parse files as that would require parsing multiple tens of GB of data in the case of MPEG TS files (Blu-rays), and that's the primary source of PGS subs. At the moment `mkvmerge` simply disregards this particular issue & doesn't care about the duration. Yes, I'm very aware that this is problematic, both with playback and operations such as splitting & appending. You could say `mkvmerge` doesn't support those fully at this time.

Reply:
I guess that ship has sailed with PGS. However I don't think a **MUST** have no duration (should be **MUST NOT** have a duration is a appropriate. If some tools want to post-mux/remux the file to add durations to help editing/seeking, we should not forbid it.

I suggest: `**SHOULD NOT** have a duration, unless it is known at the time of muxing.`

Reply:
We'll probably have to edit again when we publish it, as SHOULD and SHOULD NOT rules have to explain in which it is not the case. But that's good enough for now.